specification/compute/resource-manager/Microsoft.Compute/galleryRP/stable/2022-03-03/gallery.json

```diff
@@ -2685,11 +2863,79 @@
           }
         },
         "source": {
-          "$ref": "#/definitions/GalleryArtifactVersionSource"
+          "$ref": "#/definitions/GalleryDiskImageSource"
```

Comment:
This change is a breaking change to SDKs and it is hard for our SDKs to resolve that.
Could we change the implementation here a little bit to keep its original name?

Reply:
@ArcturusZhang 
The issue is that previously both GalleryImageVersionStorageProfile and GalleryDiskImage both refer to an object called GalleryArtifactVersionSource for the source field. This isn't really correct because they're logically different concepts.  These changes fixes this issue.

To illustrate this issue, the current swagger before these changes implies that you can use source.uri field for the GalleryImageVersionStorageProfile but that is incorrect.  The uri field only exists on the GalleryDiskImage.  This is incorrect and these changes I made here fixed this issue and prevent the issue from getting worse as these two models evove independently.

This is not really a breaking change for the customer.  If the customer where to specify the source.uri field for the GalleryImageVersionStorageProfile then the service would have rejected that request.  So, we are not breaking any customers here.